### DISCUSSION TITLE
Query prototyping environment?

### DISCUSSION THREAD

**Discussion by blevine** [prisma/prisma#9181] [Q&A]
I'm using both Knex.js and Prisma.  For my Knex queries, I've made good use of the [KNEX QueryLab tool](https://michaelavila.com/knex-querylab/).  I was wondering if anything similar exists for Prisma. If not, this seems like good functionality to add to Prisma studio. Even better would be query prototyping functionality that allows you to execute the queries against a live development DB. KNEX QueryLab just shows you what the generated SQL query would be.

**Comment by thebiglabasky** [comment]
Hi @blevine!
We have built a first version of our cloud-based product called the Prisma Data Platform. It's currently in Early Access, available at https://cloud.prisma.io
You can import your Prisma projects there, configure the connection details and access a hosted data browser, as well as a Query Console which we recently released and should do just that :)
Let us know what you think!

**Reply by blevine** [reply]
That's great!  I'll give it a try.  Unfortunately, I won't be able to put any of my actual project assets in the cloud.  I don't suppose there's any chance of providing that functionality for local/on-prem environments?

**Reply by thebiglabasky** [reply]
That's something we have on our mind but for much later as we first want to solidify the feature set. But that's good to know you're looking for more control. 
What's the reasoning behind that need? Security? Data access exposure? 

Would Cloud-based solutions based on private networks mitigate the concerns or does it have to be fully on-prem?

**Reply by blevine** [reply]
I work for a very paranoid company which is only just now considering putting some services in the cloud. So security and control are the big issues. A VPN solution probably wouldn't mitigate those concerns.  At least not for now...

**Reply by thebiglabasky** [reply]
That's good to know, thanks for clarifying. Would you be up to hop on a short call to give more context about your company and where Prisma is used? This helps gathering insights and inform our roadmap. 
If that's the case fell free to book a slot using this link: https://calendly.com/labas-prisma/45-minutes

Thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
